charts/shoot-core/components/charts/nodelocal-dns/templates/service.yaml

```
@@ -0,0 +1,20 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kube-dns-upstream
+  namespace: kube-system
+  labels:
+    k8s-app: kube-dns-upstream
+spec:
+  ports:
+    - name: dns
+      port: 53
+      protocol: UDP
+      targetPort: 8053
```

Would it make sense to use a named port here?
```suggestion
      targetPort: dns-udp
```

---

> tested the PR and also tried the migration from IPTables to IPVS. After the migration I started a new Pod and it still had the `kube-dns` clusterIP in its `resolv.conf`.
> 
> ```
> / # cat /etc/resolv.conf
> nameserver 100.64.0.10
> ```
> 
> Is this correct? I thought, in the IPVS case Pods would need to talk to `169.254.20.10` in order to leverage the node local dns cache.


Please make sure that the cluster is fully reconciled, I just tested it and indeed the pod got the cache IP.

---

I remember there were issues previously using named-ports for services, so I would vote for sticking with numerics for service target ports. We are also not using named target ports for services elsewhere AFAIK.

---

Ok, fine for me 👍